src/dcos_e2e/backends/_aws/__init__.py

```diff
@@ -0,0 +1,424 @@
+
+        stack_name = unique
+        self._aws_key_name = stack_name
+        self._ssh_private_key = self._aws_backend.create_keypair(
+            self._aws_key_name
+        )
+
+        ssh_key_path = Path(tmp_dir / 'id_rsa')
+        ssh_key_path.write_bytes(data=self._ssh_private_key.encode('utf-8'))
+        ssh_key_path.chmod(mode=stat.S_IRUSR)
+
+        cluster_size = masters + agents + public_agents
+
+        self._cf_stack = self._aws_backend.create_stack(
+            deployment_name=stack_name,
+            cluster_size=cluster_size,
+            aws_key_name=self._aws_key_name,
+        )
+
+        bootstrap_host = self._cf_stack.get_bootstrap_host()
+        cluster_hosts = self._cf_stack.get_cluster_hosts()
+
+        cluster_hosts_iter = iter(cluster_hosts)
+
+        master_hosts = list(itertools.islice(cluster_hosts_iter, masters))
+        agent_hosts = list(itertools.islice(cluster_hosts_iter, agents))
+        public_agent_hosts = list(
+            itertools.islice(cluster_hosts_iter, public_agents)
+        )
+
+        def nodes_from_hosts(hosts: Iterable[Host]) -> Set[Node]:
+            return set(
+                [
+                    Node(
+                        ip_address=IPv4Address(host.public_ip),
+                        ssh_key_path=ssh_key_path,
+                        private_ip_address=IPv4Address(host.private_ip),
+                    ) for host in hosts
+                ]
+            )
+
+        self._bootstrap_host = Node(
+            ip_address=IPv4Address(bootstrap_host.public_ip),
+            ssh_key_path=ssh_key_path,
+            private_ip_address=IPv4Address(bootstrap_host.private_ip),
+        )
+
+        self._masters = nodes_from_hosts(master_hosts)
+        self._agents = nodes_from_hosts(agent_hosts)
+        self._public_agents = nodes_from_hosts(public_agent_hosts)
+
+        self._default_ssh_user = self._aws_backend.default_ssh_user
+
+    def install_dcos_from_path(
+        self,
+        build_artifact: Path,
+        extra_config: Dict[str, Any],
+        log_output_live: bool,
+    ) -> None:
+        """
+        Args:
+            build_artifact: The `Path` to a build artifact to install DC/OS
+                from.
+            extra_config: May contain extra installation configuration
+                variables that are applied on top of the default DC/OS
+                configuration of the AWS backend.
+            log_output_live: If `True`, log output of the installation live.
+
+        Raises:
+            NotImplementedError: `NotImplementedError` because the AWS
+                backend can be utilized more efficiently by installing DC/OS
+                from from a URL using the advanced installation method.
+        """
+        message = (
+            'The AWS backend does not support the installing DC/OS from '
+            'a build artifact on the file system. This is due a more '
+            'efficient method existing in `install_dcos_from_url`.'
```

Comment:
Is the reason really that it is more efficient?

Reply:
Well, indirectly. If you create the build artifact locally then it is just as efficient.